src/server/internalServer.cpp

```diff
@@ -132,6 +132,56 @@ std::vector<T> subrange(const std::vector<T>& v, size_t s, size_t n)
 
 } // unnamed namespace
 
+std::set<std::string> InternalServer::selectBooks(const RequestContext& request, const std::string& prefix) const
```

Comment:
The meaning of `prefix` (which first appeared in the previous commit) is still mysterious here. Please describe its purpose in the commit message or in a code comment.

Reply:
Addi in commit message

> The prefix will be used to parse a "query to select book" in different context.
    For now we have only one context : selecting books for the catalog search.
    But we will want to select books to do fulltext search on them
    (will be done in later commit)